libtransmission/peer-mgr-active-requests.cc

```diff
@@ -114,7 +114,7 @@ std::vector<tr_block_index_t> ActiveRequests::remove(tr_peer const* peer)
 
     for (auto const& [block, peers_at] : impl_->blocks_)
     {
-        if (peers_at.count(peer) != 0U)
+        if (peers_at.contains(peer))
```

Comment:
Not sure why the CI passed, but `std::map::contains()` is a C++20 feature.

Speaking of which, the C++20 bump has been brought up multiple times in the past year. We might as well make the jump now, then gradually start using its features depending on the CI results. wdyt?

Reply:
> Not sure why the CI passed, but std::map::contains() is a C++20 feature.

Because this is a `small::map` instead of a `std::map` :smile_cat: 

> Speaking of which, the C++20 bump has been brought up multiple times in the past year. We might as well make the jump now, then gradually start using its features depending on the CI results. wdyt?

Personally I'd rather be using C++20 than 17. My only question is whether or not a good C++20 compiler exists on all the CI machines, e.g. the older Macs and the old distros in TeamCity's Sanity runs. But writing a PR to bump to 20 is an easy way to find out :smile_cat:

Reply:
> This is a `small::map` instead of a `std::map` which has `contains()` even in C++17

Oh.

> But writing a PR to bump to 20 is an easy way to find out 😸

Yep I'll be getting on that.

Reply:
We don't have CI for macOS 11 anymore, and I'd like to still support it.
But it's your call: if we limit our support to what GitHub Actions can support, then what is the need for the TeamCity builds maintained by Mike?

Reply:
Well, let's start with what we have: GitHub Actions support Xcode 13. Then it means that the C++20 features listed in this table with a minimum requirement of Xcode 14, 15 or 16 won't be supported:
https://developer.apple.com/xcode/cpp/#c++20

Reply:
Let's move this discussion to https://github.com/transmission/transmission/pull/7191 :+1: